src/values.rs

```diff
@@ -123,6 +171,40 @@ impl PartialOrd for Json {
     }
 }
 
+/// A data structure holding a boolean value of a type that has not
+/// been standardized yet.
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+pub struct ExtBool {
+    pub value: bool,
+
+    /// The vendor. Used for namespacing purposes, to avoid
+    /// confusing two incompatible extensions with similar
+    /// names. For instance, "foxlink@mozilla.com".
+    pub vendor: String,
```

Comment:
Side-note: as part of #11, these strings should become atoms. Otherwise, we're going to lose lots of space.